pkg/action/install.go

```diff
@@ -103,6 +103,50 @@ func NewInstall(cfg *Configuration) *Install {
 	}
 }
 
+func (i *Install) installCRDs(crds []*chart.File) error {
+	// We do these one at a time in the order they were read.
+	for _, obj := range crds {
+		// Read in the resources
+		res, err := i.cfg.KubeClient.Build(bytes.NewBuffer(obj.Data))
+		if err != nil {
+			// We bail out immediately
+			return errors.Wrapf(err, "failed to install CRD %s", obj.Name)
+		}
+		// On dry run, bail here
+		if i.DryRun {
+			i.cfg.Log("WARNING: This chart or one of its subcharts contains CRDs. Rendering may fail or contain inaccuracies.")
+			continue
+		}
+		// Send them to Kube
+		if _, err := i.cfg.KubeClient.Create(res); err != nil {
+			// If the error is CRD already exists, continue.
+			if apierrors.IsAlreadyExists(err) {
+				crdName := res[0].Name
+				i.cfg.Log("CRD %s is already present. Skipping.", crdName)
+				continue
+			}
+			return errors.Wrapf(err, "failed to instal CRD %s", obj.Name)
+		}
+
+		// Invalidate the local cache.
+		if discoveryClient, err := i.cfg.RESTClientGetter.ToDiscoveryClient(); err != nil {
+			// On error, we don't want to bail out, since this is unlikely
+			// to impact the majority of charts.
+			i.cfg.Log("Could not clear the discovery cache: %s", err)
```

Comment:
So I was thinking about this some more. I think we actually _do_ want to bail out here. If we got into this block, the user was meaning to install CRDs and we don't want their release to just fail on them. What do you think?

Reply:
Oh and also bail out on a failed wait

Reply:
I don't like the idea of failing on transient errors that are unlikely to impact the user. It seems wrong to me to take something that is not fatal (a cleared cache, a wait that is likely not going to take long anyway) and then require the user to take action.

However, I don't care enough to drag on this PR longer, so I will change it.